File: HSG_THANH_HOA_2/contestants/huynh_minh_thuan/BAI2.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    ifstream inp("BAI2.inp");
    ofstream out("BAI2.out");

    int n, k, range = 0;
    string str;
    inp >> n >> k >> str;

    for (int i = 1; i < str.size(); i++) {
        if (str[i] == str[i - 1]) range += 2;
        else range++;
    }

    while (k--) {
        int index;
        inp >> index;

        index--;

        if (index > 0) {
            if (str[index - 1] != str[index]) range++;
            else range--;
        }

        if (index < str.size() - 1) {
            if (str[index + 1] != str[index]) range++;
            else range--;
        }

        str[index] = (str[index] == '0' ? '1' : '0');

        out << range << endl;
    }

    inp.close();
    out.close();

    return 0;
}

File: HSG_HAI_DUONG_22_23/contestants/bach_do/bai2.py
with open ("bai2.inp","r") as fi:
    n=int(fi.readline())
    s=str(fi.readline())
    a=0
    b=0
    if s[0]==str(0):
        a=0        
    elif s[0]==str(1):
        a+=1
        b+=a
    for i in range(1,n):
        if s[i]==str(1):
            a+=1
            b+=a
        elif s[i]==str(0):
            a-=a
with open("bai2.out","w") as fo:
    fo.write(str(b))
fi.close()
fo.close()